src/ServerFactory.php

```diff
@@ -13,39 +13,10 @@
 use Swoole\Runtime as SwooleRuntime;
 
 use function array_replace;
-use function in_array;
 use function method_exists;
 
-use const SWOOLE_BASE;
-use const SWOOLE_PROCESS;
-use const SWOOLE_SOCK_TCP;
-use const SWOOLE_SOCK_TCP6;
-use const SWOOLE_SOCK_UDP;
-use const SWOOLE_SOCK_UDP6;
-use const SWOOLE_UNIX_DGRAM;
-use const SWOOLE_UNIX_STREAM;
-
 class ServerFactory
```

Comment:
Would it be better to remove ServerFactory completely? The request runner could then just accept a swoole server and ->set() whatever options it needs to before finally starting the swoole server

Reply:
Doing so would be a BC break, for a few reasons:

- Anybody extending the class now is no longer having their class called, which will lead to bugs.
- The constructor of the `SwooleRequestHandlerRunner` currently accepts the `ServerFactory`, which means removing it would require a signature change in the constructor of that class.

As it is, however, the `ServerFactory` now has a changed constructor, which is already a BC break.

As such, I'd argue we go full tilt, and make _all_ the changes. We can release a 2.0 almost immediately, and let users know they can adopt it immediately if they were not extending any of these factories.